api/rest/restapi.go

```diff
@@ -646,27 +646,42 @@ func (api *API) unpinHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// filterOutPin returns true if the given pin should be filtered out according
+// to the input filter type
+func (api *API) filterOutPin(filter types.FilterType, pin types.Pin) bool {
+	switch filter {
+	case types.FilterPin:
```

Comment:
I have a question: why do you need a FilterType if you already have `PinType` defined and they' re the same thing (+ `-1` meaning all)

Reply:
I thought this was what you you were asking for with "have a filter type int as a receiver" :).  I will gladly switch to strings.

Reply:
I mean, using `api.PinType`. Strings is only for the query parameter.